### ISSUE TITLE
Region Exception: NumberField cannot be converted to type

### ISSUE THREAD

**Issue by CentricStraaten** [PiranhaCMS/piranha.core#385]
I created a custom region named Carousel.

```
using Piranha.AttributeBuilder;
using Piranha.Extend.Fields;
using Piranha.Models;

namespace Synergo.Kantine.Models.Regions
{
    public class Carousel
    {
        /// <summary>
        /// Gets/sets the carousel interval.
        /// </summary>
        [Field]
        public NumberField Interval { get; set; }
    }
}
```
I added the region to the StartPage.

```
using Piranha.AttributeBuilder;
using Piranha.Models;

namespace Synergo.Kantine.Models
{
    [PageType(Title = "Start page")]
    [PageTypeRoute(Title = "Default", Route = "/start")]
    public class StartPage : Page<StartPage>
    {
        /// <summary>
        /// Gets/sets the page heading.
        /// </summary>
        [Region]
        public Regions.Heading Heading { get; set; }

        /// <summary>
        /// Gets/sets the carousel.
        /// </summary>
        [Region]
        public Regions.Carousel Carousel { get; set; }

        /// <summary>
        /// Default constructor.
        /// </summary>
        public StartPage() {

        }
    }
}
```

The region is displayed in the Manager, even if the setting is untitled.

![piranha](https://user-images.githubusercontent.com/17175831/45804689-13184a80-bcbd-11e8-97ab-18730d014187.PNG)

I can successfully save the region, but when I try to view the page an exception is thrown.

`**System.ArgumentExeption:** 'Object of type 'Piranha.Extend.Fields.NumberField' cannot be converted to type 'Synergo.Kantine.Models.Regions.Carousel'.'`

![image](https://user-images.githubusercontent.com/17175831/45805037-ff211880-bcbd-11e8-8255-4a78fc5dd05b.png)

**Comment by tidyui** [comment]
Yes, the documentation is a bit lacking here and I will update it straight away. The concept for setting up Regions & Fields is like this.

### TLDR

Single field regions does not work with custom classes.

## Regions with a single field

If you want a single field as a region, there will be **no label** shown for the field as the main region button works as the label. Also you **MUST** define the single field region like this:

    [PageType(Title = "Start page")]
    [PageTypeRoute(Title = "Default", Route = "/start")]
    public class StartPage : Page<StartPage>
    {
        [Region]
        public NumberField Carousel { get; set; }
    }

## Regions with multiple fields

When adding multiple fields to a region you do so by adding them into a custom class, just like you have done in your example.

    public class Carousel
    {
        [Field]
        public NumberField Interval { get; set; }

        [Field]
        public StringField AnotherField { get; set; }
    }

    [PageType(Title = "Start page")]
    [PageTypeRoute(Title = "Default", Route = "/start")]
    public class StartPage : Page<StartPage>
    {
        [Region]
        public Carousel Carousel { get; set; }
    }

Best regards

**Comment by CentricStraaten** [comment]
Thank you, problem solved.

**Comment by tidyui** [comment]
I updated the main Content Type wiki page here:

https://github.com/PiranhaCMS/piranha.core/wiki/Content-Type-Anatomy
